app/workers/asset_upload_state_worker.rb

```diff
@@ -0,0 +1,13 @@
+require 'services'
+
+class AssetUploadStateWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: 'low_priority'
+
+  def perform(asset_id)
+    asset = Asset.find(asset_id)
```

Comment:
I think this'll need to be `Asset.unscoped.find` to deal with deleted assets, won't it?

Reply:
Oh, yeah. Well spotted. Doh! I'm not very good at this am I!

Reply:
I've fixed this and have re-written the commits. I'm now going to rebase against `master` and force-push in preparation for merging.